pkg/apis/triggers/v1alpha1/event_listener_types.go

```diff
@@ -61,8 +63,41 @@ type TriggerTemplateRef struct {
 	APIVersion string `json:"apiversion,omitempty"`
 }
 
-// EventListenerStatus defines the observed state of EventListener
-type EventListenerStatus struct{}
+const (
+	// ServiceExists is the ConditionType set on the EventListener, which specifies Service existence.
+	ServiceExists apis.ConditionType = "Service"
+	// DeploymentAvaiable is the ConditionType set on the EventListener to reflect the actual DeploymentAvailable status.
+	DeploymentAvailable apis.ConditionType = apis.ConditionType(appsv1.DeploymentAvailable)
+	// DeploymentProgressing is the ConditionType set on the EventListener to reflect the actual DeploymentProgressing status.
+	DeploymentProgressing apis.ConditionType = apis.ConditionType(appsv1.DeploymentProgressing)
```

Comment:
Do we need to account for `DeploymentReplicaFailure`? (https://godoc.org/k8s.io/api/apps/v1#DeploymentConditionType)